site/book/05-developing-functions/02-developing-in-Go.md

```diff
@@ -13,121 +12,161 @@ the following features:
   manipulating comments; however, it comes at the cost of complexity compared to
   representing resources as idiomatic data structures.
 
+## Prerequisites
+
+#### Install kpt
+
+`kpt` is used to get the example Go projects "get-started" and provides an easy way to generate the function specification
+object `ResourceList`.
+
+[kpt.dev/installation](https://kpt.dev/installation/)
+
+#### Install Docker
+
+The KRM function can be released as a docker or OCI image and used by `kpt fn`.
+It can also be executed as a binary. 
+This guide uses Docker to build the image. It also shows the executable approach.
+
+[docker installation](https://docs.docker.com/get-docker/)
+
 ## Quickstart
 
-### Create the go module
+### Initialize your project
+
+We start from a "get-started" package which contains a `main.go` file with some scaffolding code.
 
 ```shell
-$ go mod init github.com/user/repo; go get sigs.k8s.io/kustomize/kyaml@v0.12.0
-```
+export FUNCTION_NAME=<your KRM function name>
 
-### Create the `main.go`
+# Get the "get-started" package.
+kpt pkg get https://github.com/GoogleContainerTools/kpt-functions-sdk.git/go/get-started@master ${FUNCTION_NAME}
 
-This is a simple function that adds the annotation `myannotation` with the
-provided value:
+cd ${FUNCTION_NAME}
+
+# Initialize the Go module
+go mod init
+go mod tidy
+```
+
+### Write the KRM function logic
+ 
+Take a look at the `main.go` (as below) and complete the `Run` function.
 
 ```go
 // main.go
 package main
 
 import (
-	"os"
+  "fmt"
+  "os"
 
-	"sigs.k8s.io/kustomize/kyaml/fn/framework"
-	"sigs.k8s.io/kustomize/kyaml/fn/framework/command"
-	"sigs.k8s.io/kustomize/kyaml/kio"
-	"sigs.k8s.io/kustomize/kyaml/yaml"
+  "github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
 )
 
+// EDIT THIS FUNCTION!
+// This is the main logic. rl is the input `ResourceList` which has the `FunctionConfig` and `Items` fields.
+// You can modify the `Items` and add result information to `rl.Results`.
+func Run()(rl *fn.ResourceList) (bool, error) {
+// Your code
+}
```

Comment:
I liked the previous example was showing `how to set annotations` as a minimal example. It also helps in showing the features `fn` library provides without talking about.